File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOHeartRateModeModel.kt
//
//  IDOHeartRateModeModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Set Heart Rate Mode Event

open class IDOHeartRateModeModel(
    mode: Int,
    hasTimeRange: Int,
    startHour: Int,
    startMinute: Int,
    endHour: Int,
    endMinute: Int,
    measurementInterval: Int
) : IDOBaseModel {

    /// 0: Turned Off
    /// 1: Manual Mode
    /// 2: Automatic
    /// 3: Continous Monitoring
    /// -1:Invalid
    @SerializedName("mode")
    var mode: Int = mode

    /// Whether there is a time range 0: No 1: Yes
    @SerializedName("has_time_range")
    var hasTimeRange: Int = hasTimeRange

    @SerializedName("start_hour")
    var startHour: Int = startHour

    @SerializedName("start_minute")
    var startMinute: Int = startMinute

    @SerializedName("end_hour")
    var endHour: Int = endHour

    @SerializedName("end_minute")
    var endMinute: Int = endMinute

    /// Measurement interval (in minutes)
    @SerializedName("measurement_interval")
    var measurementInterval: Int = measurementInterval


    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOHeartRateModeModel.swift
//
//  IDOHeartRateModeModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Set Heart Rate Mode Event
@objcMembers
public class IDOHeartRateModeModel: NSObject, IDOBaseModel {
    /// 0: Turned Off
    /// 1: Manual Mode
    /// 2: Automatic
    /// 3: Continous Monitoring
    /// -1:Invalid
    public var mode: Int
    /// Whether there is a time range 0: No 1: Yes
    public var hasTimeRange: Int
    public var startHour: Int
    public var startMinute: Int
    public var endHour: Int
    public var endMinute: Int
    /// Measurement interval (in minutes)
    public var measurementInterval: Int
    
    enum CodingKeys: String, CodingKey {
        case mode
        case hasTimeRange = "has_time_range"
        case startHour = "start_hour"
        case startMinute = "start_minute"
        case endHour = "end_hour"
        case endMinute = "end_minute"
        case measurementInterval = "measurement_interval"
    }
    
    public init(mode: Int, hasTimeRange: Int, startHour: Int, startMinute: Int, endHour: Int, endMinute: Int, measurementInterval: Int) {
        self.mode = mode
        self.hasTimeRange = hasTimeRange
        self.startHour = startHour
        self.startMinute = startMinute
        self.endHour = endHour
        self.endMinute = endMinute
        self.measurementInterval = measurementInterval
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

